loki/analyse/analyse_dataflow.py

```diff
@@ -269,12 +269,15 @@ def visit_Deallocation(self, o, **kwargs):
     visit_Nullify = visit_Deallocation
 
     def visit_Import(self, o, **kwargs):
-        defines = self._symbols_from_expr(o.symbols or ())
+        defines = set(s.clone(dimensions=None) for s in FindTypedSymbols().visit(o.symbols or ())
+                      if isinstance(s, ProcedureSymbol))
```

Comment:
Let's stick to the commonly used abstraction:
```suggestion
        defines = self._symbols_from_expr(o.symbols or (), condition=lambda s: isinstance(s, ProcedureSymbol))
```

Reply:
Ah I remember now. I intentionally used `FindTypedSymbols` here because a `ProcedureSymbol` is not a `Variable` and so it's not picked up by `FindVariables`.